lib/spack/spack/build_systems/cmake.py

```diff
@@ -83,14 +83,21 @@ def build_type(self):
         return 'RelWithDebInfo'
 
     @property
-    def root_cmakelists_dir(self):
-        """Returns the location of the root CMakeLists.txt
+    def cmakelists_directory(self):
+        """The relative path to the directory containing CMakeLists.txt
```

Comment:
"relative to the root directory of the extracted tarball"? I wasn't sure exactly what to call it. But I agree, the whole point of this PR is to make things clearer and more intuitive.